client/src/components/PostPreview.tsx

```diff
@@ -10,13 +10,45 @@ import Typography from "@mui/material/Typography";
 import GenerateTags from "./Tags";
 import ViewPostDialog from "./ViewPostDialog";
 
-import { PostUser } from "../api/v1/index";
+import { PostUserPreview } from "../api/v1/index";
 
+  /**
+   * Return a string of the date relative to now in minutes, hours or days. If the date
+   * is over a month, return the date stringified. 
+   * 
+   * @param base - The base date being compared with
+   * @param date - The date converted into relative time
+   * @returns String containing the date relative to now
+   */
+function relativeTime(date: Date) {
```

Comment:
There should be libraries which do this for us. Benefit would be that all cases are handled and they are tested.

Reply:
I added some tests for this